### PR TITLE
✨ Generalize heder, fix adding custom language

### LINKED ISSUE #3: Move LanguageSwitcher to the theme
Invent how to generalize `LanguageSwitcher` to be able to move into the theme. Maybe use "render prop" pattern to allow custom rendering and thus allow user render the switcher however he wants.

### PR DESCRIPTION
**Pull request by filip-dot-cube**
Generalize header. 

When testing I've noticed that adding another language was impossible. 

This therefore also resolves #3 

Had to fix it by passing languages from gatsby-config. Languages still have to be shadowed for there are absolute imports that I cant easily override. Also passing ymls directly is impossible since gastby-config runs sooner than gatsby-node where webpack loaders can be set.

### CODE REVIEW

File: gatsby-theme-localized/gatsby-node.js [hunk 1]
```diff
@@ -33,17 +34,4 @@ exports.onCreatePage = ({ page, actions }) => {
     });
 };
 
-exports.onCreateWebpackConfig = ({ stage, rules, loaders, plugins, actions }) => {
```
Comment (horaklukas) [inline gatsby-theme-localized/gatsby-node.js]: Please revert this, we need it

File: gatsby-theme-localized/src/components/LanguageSwitcher/LanguageSwitcher.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        {({ originalPath, locale, languages }) => {
+            return(
+                <div>
+                {languages.map(lang => {
+                    const isCurrent = lang.locale === locale;
+                    const path = getLocalizedPath(originalPath, lang.locale, languages)
+
+                    return render({lang, isCurrent, path})
+                })}
+             </div>
```
Comment (horaklukas) [inline gatsby-theme-localized/src/components/LanguageSwitcher/LanguageSwitcher.jsx]: `div` is not properly aligned, do u use Prettier? 🤔

File: gatsby-theme-localized/src/components/LanguageSwitcher/LanguageSwitcher.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        )}}
+    </PageContext.Consumer>
+    );   
+} 
+
+export default LanguageSwitcher;
```
Comment (horaklukas) [inline gatsby-theme-localized/src/components/LanguageSwitcher/LanguageSwitcher.jsx]: we need to define `PropTypes` and specify if `render` is required or not. My opinion is that it should not be require and define default `render`, eg.

```js
const defaultRender = ({ lang, isCurrent, path }) => 
  isCurrent ? 
   (	
    <span key={lang.locale}>{lang.label}</span>	
   ) : (	
     <Link key={lang.locale} to={path}>	
        <span>{lang.label}</span>	
      </Link>	
   )
```

File: site/src/components/LanguageSwitcher/LanguageSwitcher.jsx [hunk 4]
```diff
@@ -1,33 +1,22 @@
 import React from 'react'
 import { Link } from 'gatsby'
 
-import { languages, getLocalizedPath } from "../../gatsby-theme-localized/i18n";
+import { LanguageSwitcher as Switcher } from "gatsby-theme-localized"
 
-import { PageContext } from "gatsby-theme-localized/src/pageContext";
+const customSwitcherElement = ({ lang, isCurrent, path }) => {
```
Comment (horaklukas) [inline site/src/components/LanguageSwitcher/LanguageSwitcher.jsx]: I would personally renamed it to `renderCustomSwitcher` 👌

Comment (filip-dot-cube) [reply site/src/components/LanguageSwitcher/LanguageSwitcher.jsx]: 🧐

**Pull request merged**